graphql/resolve/mutation_rewriter.go

```diff
@@ -587,14 +589,14 @@ func (urw *UpdateRewriter) Rewrite(
 
 	upsertQuery := RewriteUpsertQueryFromMutation(m, authRw, MutationQueryVar, m.Name(), "")
 	srcUID := MutationQueryVarUID
-
-	if setArg == nil && delArg == nil {
+	objDel, ok := delArg.(map[string]interface{})
+	if setArg == nil && (delArg == nil || (len(objDel) == 0 && ok)) {
```

Comment:
I believe we can return at this stage even for an empty set. Let's compute objSet here and return if len(objSet) == 0 as well.